### ISSUE TITLE
[BUG] Mailbox::switchMailbox ends in wrong url

### ISSUE THREAD

**Issue by ScreamingDev** [barbushin/php-imap#372]
**Describe the bug**
**To Reproduce**

see https://github.com/barbushin/php-imap/blob/master/src/PhpImap/Mailbox.php#L370

The used code:
```php
$mailbox = new Mailbox(...);
$mailbox->switchMailbox('{foo.bar/imap/tls}Baz'); // ends in "{foo.bar/imap/tls}{foo.bar/imap/tls}Baz"
$mailbox->switchMailbox('Baz'); // ends in "BBaz"
```

Due to `vendor/php-imap/php-imap/src/PhpImap/Mailbox.php:370` the imap url is always wrong no matter if given relative or absolute.

**Comment by Sebi94nbg** [comment]
Line 370 saves only the latest set IMAP path to the global variable: `$this->imapPath = $imapPath;`

This can not be the issue. There might be an issue with the function `getCombinedPath()`. I'll check this.

**Comment by ScreamingDev** [comment]
No. Look at it. Might be meant like

```
$this->imapPath = $this->getCombinedPath($imapPath, true);
        $this->imap('reopen', $this->imapPath);
```

or similar. `switchMailbox` is the problem

**Comment by pscheit** [comment]
I can confirm this bug.

Its indeed like @ScreamingDev shows. $this->imapPath should not be set, before getCombinedPath()  is called

**Comment by davidgv88** [comment]
I have the same problem

**Comment by davidgv88** [comment]
Quick Fix

```php
public function switchMailbox($imapPath)
    {
        $this->imapPath = $imapPath;
        //$this->imap('reopen', $this->getCombinedPath($imapPath, true));
        $this->imap('reopen', $imapPath);
    }
```

**Comment by Sebi94nbg** [comment]
Closing this since a fix (PR #389) has been merged into `develop` branch. The next version will include this fix.

Please comment or reopen this issue, if you still face this issue.
